### DISCUSSION TITLE
Did version 1.1.7 change themes?

### DISCUSSION THREAD

**Discussion by Aetherinox** [mayurankv/Obsidian-Code-Styler#226] [General]
With the release of version 1.1.7, I noticed my themes were reset.

I had a backup of my JSON with the settings I had picked. So I closed Obsidian, overwrite the `data.json` file, re-launched, and the theme was still not there. Yet it was in the json file.

**Comment by mayurankv** [comment]
It hasn't changed, that's surprising. Would you mind sharing the data.json? Sorry about that! (There is some logic which should auto-update the settings, there was an issue with that but if you update now, it should all be working).

**Comment by mayurankv** [comment]
This is definitely a bug, so I'm moving this to an issue

**Comment by Aetherinox** [comment]
Sure, attached is the backed up theme file from before the update.

[data.json](https://github.com/mayurankv/Obsidian-Code-Styler/files/14390666/data.json)

I tried roughly 3 times to close out Obsidian and re-write the backed up file. When I'd reload the code plugin, the only theme in the dropdown was:
- Default
- Solarized

I finally got it to work, but I had to take my backed up json file, open it in VSCode, and then go through each setting one by one and track down where in the settings interface that setting was associated with, and then set the new value manually.

I figured you just changed a lot, because if I look at the bottom of the new and old data.json, my backed up one reads
- `"version": "1.0.11"`

The new one reads
- `"version": "1.1.7"`

**Reply by mayurankv** [reply]
Glad to hear you got it working, sorry about the annoying workaround, I'm aware the settings files are pretty large as well.

**Comment by Aetherinox** [comment]
It's alright. I'm a developer, I get random stuff happening. Just figured I'd mention in case it wasn't due to any changes.
